tests/test_router_ospf_area_vl.rb

```diff
@@ -0,0 +1,305 @@
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/router_ospf'
+require_relative '../lib/cisco_node_utils/router_ospf_vrf'
+require_relative '../lib/cisco_node_utils/router_ospf_area_vl'
+
+# TestRouterOspfAreaVirtualLink - Minitest for RouterOspfAreaVirtualLink
+# node utility class
+class TestRouterOspfAreaVirtualLink < CiscoTestCase
+  @skip_unless_supported = 'ospf_area_vl'
+  @@pre_clean_needed = true # rubocop:disable Style/ClassVars
+
+  def setup
+    super
+    remove_all_ospfs if @@pre_clean_needed
+    @@pre_clean_needed = false # rubocop:disable Style/ClassVars
+  end
+
+  def teardown
+    remove_all_ospfs
+    config 'no feature ospf'
```

Comment:
I don't think you need this feature disablement. It just slows down the test. If you really want it then I would make it part of the pre clean only.